particles/Services/particles/js/ImageClassifier.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+defineParticle(({DomParticle, log, html, resolver}) => {
+
+  const template_ = html`
+<div>
+  <h2>Classification with a generic image classifier</h2>
+  <img style="max-width: 240px;" src="{{imageUrl}}" alt="Image to classify"><br>
+  <div id="progress-bar">{{progress}}</div>
+  <div>
+    <div>Label: </span><span>{{label}}</div>
+    <div>Confidence: </span><span>{{probability}}</div>
+  </div>
+</div>
+  `;
+
+  const url = resolver(`ImageClassifier/../../assets/waltbird.jpg`);
+  const modelUrl = 'https://tfhub.dev/google/imagenet/mobilenet_v1_100_224/classification/1';
+  const labelUrl = resolver(`ImageClassifier/../../assets/ImageNetLabels.txt`);
+
+  return class extends DomParticle {
+    get template() {
+      return template_;
+    }
+
+    update({}, state) {
+      // TODO(sjmiles): update() is called during SpecEx, while
+      // render() is not. We'll put our processing code in render()
+      // to avoid being expensive at SpecEx time.
+    }
+
+    render({}, state) {
+      // formerly update
+      if (!state.loaded) {
+        state.loaded = true;
+        this.loadModel(modelUrl, {fromTFHub: true});
```

Comment:
loadModel is async but we're not awaiting it, is that ok? Also @sjmiles do we need to do the `start/endBusy` thing within particle logic if we've async logic?

Reply:
As I understand it, we want to start work that will be completed asynchronously (with `loadModel` and `getModelLabels`). They don't need to be finished in this function or in any order.

We do need to make sure we don't call these functions in an infinite loop, which we are controlling via flags (`state.loaded`, `state.labels`). Yes, this is recursive, but it has an immediate base case.

Reply:
`update` and `render` should never await anything, because they are re-entrant and it creates confusion. Alex is doing this properly (as you indicated Walt) in the sense that update kicks off async work, and the async work changes state. This is really what the 'state update' concept exists: there are N asynchronous signals to coordinate.
Re: `startBusy`, utility or generic functions are required to bracket async in `start/endBusy` calls because they do not know how they will be used. Particles themselves only need use those brackets if they want to ensure some work gets done (like setting descriptions) before SpecEx is complete (and we have dodged SpecEx completely in these Particles).